extensions/amp-springboard-player/0.1/amp-springboard-player.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright 2015 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {loadPromise} from '../../../src/event-helper';
+import {setStyles} from '../../../src/style';
+
+class AmpSpringboardPlayer extends AMP.BaseElement {
+
+  /** @override */
+  createdCallback() {
+    this.preconnect.url('https://cms.springboardplatform.com');
+    this.preconnect.url('https://www.springboardplatform.com');
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    const width = this.element.getAttribute('width');
+    const height = this.element.getAttribute('height');
+    const mode = AMP.assert(
+        this.element.getAttribute('data-mode'),
+        'The data-mode attribute is required for <amp-springboard-player> %s',
+        this.element);
+    const contentId = AMP.assert(
+        this.element.getAttribute('data-content-id'),
+        'The data-content-id attribute is required for' +
+        '<amp-springboard-player> %s',
+        this.element);
+    const domain = AMP.assert(
+        this.element.getAttribute('data-domain'),
+        'The data-domain attribute is required for <amp-springboard-player> %s',
+        this.element);
+
+    this.width_ = width;
```

Comment:
Add docs for these? Usually something like:

```
/** @private @const {number} */
this.width_ = width;

/** @private @const {string} */
this.mode_ = mode;
```

These are not enforced by the linter so far but I believe at some point they will.